src/NumericalAlgorithms/Interpolation/CubicSpline.hpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_spline.h>
+#include <memory>
+#include <vector>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A natural cubic spline interpolation class
+ *
+ * The class builds a cubic spline interpolant with natural boundary conditions
+ * using the `x_values` and `y_values` passed into the constructor. For details
+ * on the algorithm see the GSL documentation on `gsl_interp_cspline`.
+ *
+ * Here is an example how to use this class:
+ *
+ * \snippet Test_CubicSpline.cpp interpolate_example
+ */
+class CubicSpline {
+ public:
+  CubicSpline(std::vector<double> x_values,
+              std::vector<double> y_values) noexcept;
+
+  CubicSpline() noexcept = default;
+  CubicSpline(const CubicSpline& /*rhs*/) = delete;
+  CubicSpline& operator=(const CubicSpline& /*rhs*/) = delete;
+  CubicSpline(CubicSpline&& /*rhs*/) noexcept = default;
+  CubicSpline& operator=(CubicSpline&& rhs) noexcept = default;
+  ~CubicSpline() noexcept = default;
+
+  double operator()(double x_to_interp_to) const noexcept;
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  struct gsl_interp_accel_deleter {
+    void operator()(gsl_interp_accel* const acc) const noexcept;
```

Comment:
I think clang-tidy will complain that you have a const in a non-definition here.